src/components/SearchForm/SearchForm.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react";
+import "./SearchForm.css";
+
+const SearchForm = ({ handleSearch }) => { 
+  const [query, setQuery] = useState("");
+  const [error, setError] = useState("");
+
+  // Handle form submission
+  const onSubmit = (e) => {
+    e.preventDefault();
+    if (!query.trim()) {
+      setError("Please enter a keyword");
+    } else {
+      setError("");
+      handleSearch(query);  // Pass the search query to App.jsx
```

Comment:
`setQuery("");`


You need to **reset** the form only when you **open** it or better in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.